From 237e88dcbcb0098e1a8a0887fb7299fcf18650a5 Mon Sep 17 00:00:00 2001
From: Koen Vandeputte <koen.vandeputte@ncentric.com>
Date: Mon, 18 Mar 2019 17:39:26 +0100
Subject: [PATCH] media: i2c: tda1997x: select V4L2_FWNODE

Building tda1997x fails now unless V4L2_FWNODE is selected:

drivers/media/i2c/tda1997x.o: in function `tda1997x_parse_dt'
undefined reference to `v4l2_fwnode_endpoint_parse'

While at it, also sort the selections alphabetically

Fixes: 9ac0038db9a7 ("media: i2c: Add TDA1997x HDMI receiver driver")
Signed-off-by: Koen Vandeputte <koen.vandeputte@ncentric.com>
Cc: Akinobu Mita <akinobu.mita@gmail.com>
Cc: Bingbu Cao <bingbu.cao@intel.com>
Cc: Hans Verkuil <hverkuil-cisco@xs4all.nl>
Cc: Jacopo Mondi <jacopo+renesas@jmondi.org>
Cc: Matt Ranostay <matt.ranostay@konsulko.com>
Cc: Mauro Carvalho Chehab <mchehab@kernel.org>
Cc: Robin Leblon <robin.leblon@ncentric.com>
Cc: Rui Miguel Silva <rui.silva@linaro.org>
Cc: Sakari Ailus <sakari.ailus@linux.intel.com>
Cc: Tim Harvey <tharvey@gateworks.com>
Cc: linux-kernel@vger.kernel.org
Cc: stable@vger.kernel.org # v4.17+
---
 drivers/media/i2c/Kconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/media/i2c/Kconfig
+++ b/drivers/media/i2c/Kconfig
@@ -60,8 +60,9 @@ config VIDEO_TDA1997X
 	tristate "NXP TDA1997x HDMI receiver"
 	depends on VIDEO_V4L2 && I2C && VIDEO_V4L2_SUBDEV_API
 	depends on SND_SOC
-	select SND_PCM
 	select HDMI
+	select SND_PCM
+	select V4L2_FWNODE
 	---help---
 	  V4L2 subdevice driver for the NXP TDA1997x HDMI receivers.
 
